IdentityCore/src/cache/mac/MSIDMacKeychainTokenCache.m

```diff
@@ -256,6 +257,8 @@ - (nullable instancetype)initWithGroup:(nullable NSString *)keychainGroup
             return nil;
         }
 
+        self.synchronizationQueue = dispatch_queue_create("com.microsoft.msidmackeychaintokencache", DISPATCH_QUEUE_CONCURRENT);
```

Comment:
There was a concern that this will become a bottleneck for read operations. Often, apps are executing quite many reads on multiple threads, so I requested Dean to check Apple's keychain code and see if it's necessary to have reads also serialized, or we can live with only writes being serialized. 

This code currently allows running multiple reads in parallel, but only one update and ensures reads don't run at the same time as writes.

Reply:
Also, we intend to ask Apple to clarify whether concurrent reads are OK.